### PR TITLE
feat: implement `bigframes.bigquery.json_extract_array`

### PR DESCRIPTION
**Pull request by shobsi**
This is needed to implement support for array return types in remote functions.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/operations/__init__.py [hunk 1]
```diff
@@ -652,6 +652,23 @@ def output_type(self, *input_types):
...
+    json_path: str
+
+    def output_type(self, *input_types):
+        input_type = input_types[0]
+        if not dtypes.is_json_like(input_type):
+            raise TypeError(
+                "Input type must be an valid JSON object or JSON-formatted string type."
+                + f" Received type: {input_type}"
+            )
+        pd.ArrowDtype(
```
Comment (chelsea-lin) [inline bigframes/operations/__init__.py:667]: Add a "return" could fix the doctest failure.

File: tests/system/small/bigquery/test_json.py [hunk 2]
```diff
@@ -139,3 +139,14 @@ def test_json_extract_from_string():
 def test_json_extract_w_invalid_series_type():
```
Comment (chelsea-lin) [inline tests/system/small/bigquery/test_json.py:139]: please add a similar test to improve the code coverage

**Review by chelsea-lin — APPROVED** [review]
LGTM if tests are passed and left small comments can be addressed.

File: bigframes/operations/__init__.py (hunk 1 above)
Comment (shobsi) [reply bigframes/operations/__init__.py]: Thank you, done!

File: tests/system/small/bigquery/test_json.py (hunk 2 above)
Comment (shobsi) [reply tests/system/small/bigquery/test_json.py:139]: done

**Pull request merged**